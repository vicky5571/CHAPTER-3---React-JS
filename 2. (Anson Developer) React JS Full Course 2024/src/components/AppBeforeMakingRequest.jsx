import { useState, useEffect } from "react";
import { UserDetails } from "./components/UserDetails";

export default function App() {
  const [username, setUsername] = useState("");
  const [email, setEmail] = useState("");
  const [counter, setCounter] = useState(3);
  const [sync, setSync] = useState(false);
  const [users, setUsers] = useState([
    {
      id: 1,
      username: "vicky",
      email: "vicky@gmail.com",
    },
    {
      id: 2,
      username: "Huang Xing",
      email: "huangxing@gmail.com",
    },
  ]);

  useEffect(() => {
    console.log("Rendering...");
    document.title = "React Tutorial" + counter;
  }, [sync]);

  useEffect(() => {
    const controller = new AbortController();
    controller.signal;
    async function fetchUsers() {
      try {
        const response = await fetch("https://jsonplaceholder.typicode.com/users", { signal: controller.signal });
        const json = await response.json();
        console.log(json);
      } catch (err) {
        console.log(err);
      }
    }
    fetchUsers();
    return () => {
      controller.abort();
    };
  });

  return (
    <div>
      <div>
        <div>You clicked {counter} times</div>
        <button onClick={() => setCounter((count) => count + 1)}>Click Me</button>
        <button onClick={() => setSync((current) => !current)}>Sync</button>
      </div>
      <br />
      <br />
      <form
        onSubmit={(e) => {
          e.preventDefault();
          const newUser = {
            id: counter,
            username,
            email,
          };
          setCounter((currentCounter) => currentCounter + 1);
          setUsers((currentUsersState) => [...currentUsersState, newUser]);
        }}
      >
        <div>
          <label htmlFor="username">Username: </label>
          <input
            name="username"
            type="text"
            id="username"
            value={username}
            onChange={(e) => {
              setUsername(e.target.value);
            }}
          />
        </div>

        <div>
          <label htmlFor="email">Email: </label>
          <input
            name="email"
            type="email"
            id="username"
            value={email}
            onChange={(e) => {
              setEmail(e.target.value);
            }}
          />
        </div>
        <button>Add User</button>
      </form>
      <br />
      <br />
      <br />
      {users.map((user) => (
        <UserDetails key={user.id} user={user} setUsers={setUsers} />
      ))}
    </div>
  );
}
