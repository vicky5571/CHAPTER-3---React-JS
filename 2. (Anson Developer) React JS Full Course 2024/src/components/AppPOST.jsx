import { useState } from "react";

export default function App() {
  const [blogPostData, setBlogPostData] = useState({
    title: "",
    body: "",
  });

  return (
    <div>
      <form
        onSubmit={(e) => {
          e.preventDefault();
          if (blogPostData.title && blogPostData.body) {
            fetch("https://jsonplaceholder.typicode.com/posts", {
              method: "POST",
              body: JSON.stringify({
                useId: 1,
                title: blogPostData.title,
                body: blogPostData.body,
              }),
              headers: {
                "Content-type": "application/json; charset=UTF-8",
              },
            })
              .then((response) => response.json())
              .then((data) => {
                console.log("Success!");
                console.log(data);
              })
              .catch((err) => console.log(err));
          }
        }}
      >
        <div>
          <label htmlFor="title">Title</label>
          <input
            type="text"
            id="title"
            value={blogPostData.title}
            onChange={(e) => {
              setBlogPostData((currentBlogPostData) => ({
                ...currentBlogPostData,
                title: e.target.value,
              }));
            }}
          />
        </div>
        <div>
          <label htmlFor="body">Body</label>
          <input
            type="text"
            id="body"
            value={blogPostData.body}
            onChange={(e) => {
              setBlogPostData((currentBlogPostData) => ({
                ...currentBlogPostData,
                body: e.target.value,
              }));
            }}
          />
        </div>
        <button>Create Post</button>
      </form>
    </div>
  );
}
