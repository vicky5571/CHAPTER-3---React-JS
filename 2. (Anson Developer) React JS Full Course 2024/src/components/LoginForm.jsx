import { useEffect } from "react";
export function LoginForm() {
  useEffect(() => {
    const resizeEventHandler = (e) => {
      console.log("Window/ViewPort Resized!");
    };
    window.addEventListener("resize", resizeEventHandler);

    return () => {
      console.log("Unmounting LoginForm");
      console.log("Removing Resize Event Listener");
      window.removeEventListener("resize", resizeEventHandler);
    };
  }, []);
  return (
    <form
      onSubmit={(e) => {
        e.preventDefault();
        const formData = new FormData(e.target);
        const username = formData.get("username");
        const password = formData.get("password");
        fetch("localhost:5173/api/login"),
          {
            body: {
              username,
              password,
            },
            method: "POST",
          };
      }}
    >
      <label htmlFor="username">Username</label>
      <br />
      <input
        id="username"
        type="text"
        onChange={(e) => {
          console.log(`Username: ${e.target.value}`);
        }}
        name="username"
      />
      <br />
      <label htmlFor="password">Password</label>
      <br />
      <input
        id="password"
        type="password"
        onChange={(e) => {
          console.log(`Password: ${e.target.value}`);
        }}
        name="password"
      />
      <br />
      <button>Login</button>
    </form>
  );
}
