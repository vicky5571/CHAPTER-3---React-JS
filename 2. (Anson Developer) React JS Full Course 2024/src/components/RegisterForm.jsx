import { useState } from "react";

export function RegisterForm() {
  // use array destructuring
  const [formFields, setFormFields] = useState({
    username: "",
    password: "",
    displayName: "",
  });

  return (
    <form>
      <div>
        <label htmlFor="username">Username</label>
        <input id="username" value={formFields.username} onChange={(e) => setFormFields((currentState) => ({ ...currentState, username: e.target.value }))} />
      </div>
      <div>
        <label htmlFor="password">Dassword</label>
        <input id="password" value={formFields.password} onChange={(e) => setFormFields((currentState) => ({ ...currentState, password: e.target.value }))} />
      </div>
      <div>
        <label htmlFor="displayName">Display Name</label>
        <input id="displayName" value={formFields.displayName} onChange={(e) => setFormFields((currentState) => ({ ...currentState, displayName: e.target.value }))} />
      </div>
      <div>
        <span>Username: {formFields.username}</span>
      </div>
      <div>
        <span>Password: {formFields.password}</span>
      </div>
      <div>
        <span>Display Name: {formFields.displayName}</span>
      </div>
      {/* <button disabled={isDisabled}>Sign Up</button> */}
    </form>
  );
}
