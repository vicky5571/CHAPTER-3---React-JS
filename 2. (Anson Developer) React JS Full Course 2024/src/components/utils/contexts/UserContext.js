import { createContext } from "react";

export const UserContext = createContext({
  id: 0,
  username: "",
  email: "",
  name: "",
  setUserData: () => {},
});
