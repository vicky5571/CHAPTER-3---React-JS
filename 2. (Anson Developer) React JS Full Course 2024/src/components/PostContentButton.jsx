import { useContext } from "react";
import { UserContext } from "./utils/hooks/contexts/UserContext";

export function PostContentButton() {
  const { id, setUserData } = useContext(UserContext);
  return (
    <div>
      <div>
        <span>PostContentButton</span>
      </div>
      {id}
      <br />
      <button
        onClick={() => {
          setUserData((currentState) => ({ ...currentState, name: "Updated Name" }));
        }}
      >
        Click Me
      </button>
    </div>
  );
}
