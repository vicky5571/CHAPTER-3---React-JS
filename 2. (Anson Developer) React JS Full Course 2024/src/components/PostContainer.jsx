/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import { useContext, useState } from "react";
import { PostContent } from "./PostContent";
import { UserContext } from "./utils/contexts/UserContext";

export function PostContainer() {
  const [data, setData] = useState("hellow");
  const userContextData = useContext(UserContext);

  return (
    <div>
      <div>
        <span>PostContainer</span>
      </div>
      <div>Display Name: {userContextData.name}</div>
      <div>ID: {userContextData.id}</div>
      <div>Email: {userContextData.email}</div>
      <div>Username: {userContextData.username}</div>
      {/* <PostContent data={data} /> */}
    </div>
  );
}
