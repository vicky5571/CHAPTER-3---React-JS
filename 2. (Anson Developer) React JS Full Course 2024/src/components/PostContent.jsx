import { useContext } from "react";
import { UserContext } from "./utils/contexts/UserContext";
import { PostContentButton } from "./PostContainer";

export function PostContent() {
  const userContextData = useContext(UserContext);

  return (
    <div>
      <div>
        <h3>PostContent</h3>
      </div>
      <div>
        <PostContentButton />
        {userContextData.email}
      </div>
    </div>
  );
}
