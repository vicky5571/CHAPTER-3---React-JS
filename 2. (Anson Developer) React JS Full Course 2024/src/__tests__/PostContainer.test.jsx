import { describe, it, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { PostContainer } from "../components/PostContainer";
import { UserContext } from "../components/utils/contexts/UserContext";

describe("render context values", () => {
  const mockUserContextData = {
    id: 1001,
    username: "johnny",
    email: "johnny@gmail.com",
    name: "Johnny",
    setUserData: () => {},
  };
  it("should match snapshot", () => {
    const { container } = render(
      <UserContext.Provider value={mockUserContextData}>
        <PostContainer />
      </UserContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should display correct username", () => {
    render(
      <UserContext.Provider value={mockUserContextData}>
        <PostContainer />
      </UserContext.Provider>
    );

    expect(screen.getByText("Username: johnny")).toBeInTheDocument();
  });
});
