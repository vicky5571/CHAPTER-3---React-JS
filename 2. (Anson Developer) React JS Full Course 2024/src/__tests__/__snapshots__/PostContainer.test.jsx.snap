// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render context values > should display correct username 1`] = `
<div>
  <div>
    <div>
      <span>
        PostContainer
      </span>
    </div>
    <div>
      Display Name: 
      Johnny
    </div>
    <div>
      ID: 
      1001
    </div>
    <div>
      Email: 
      johnny@gmail.com
    </div>
    <div>
      Username: 
      johnny
    </div>
  </div>
</div>
`;

exports[`render context values > should match snapshot 1`] = `
<div>
  <div>
    <div>
      <span>
        PostContainer
      </span>
    </div>
    <div>
      Display Name: 
      Johnny
    </div>
    <div>
      ID: 
      1001
    </div>
    <div>
      Email: 
      johnny@gmail.com
    </div>
    <div>
      Username: 
      johnny
    </div>
  </div>
</div>
`;
